ethcore/src/client/client.rs

```diff
@@ -1843,6 +1843,12 @@ impl BlockChainClient for Client {
 				_ => return Err(filter.to_block.clone()),
 			};
 
+			// If from is greater than to, then the current bloom filter behavior is to just return empty
+			// result. There's no point to continue here.
+			if from > to {
+				return Err(filter.to_block.clone());
```

Comment:
The error returned from the RPC is `"One of the blocks specified in filter (fromBlock, toBlock or blockHash) cannot be found"`, which seems confusing because the from and to blocks can be found, they are just out of order.